app/models/trainee.rb

```diff
@@ -125,7 +125,11 @@ class Trainee < ApplicationRecord
 
   scope :ordered_by_date, -> { order(updated_at: :desc) }
   scope :ordered_by_last_name, -> { order(last_name: :asc) }
-  scope :with_subject, ->(subject) { where("LOWER(course_subject_one) = :subject OR LOWER(course_subject_two) = :subject OR LOWER(course_subject_three) = :subject", subject: subject.downcase) }
+  scope :with_subject_or_allocation_subject, lambda { |subject|
+    joins("LEFT JOIN subject_specialisms AS specialism ON specialism.name = trainees.course_subject_one OR specialism.name = trainees.course_subject_two OR specialism.name = trainees.course_subject_three")
```

Comment:
Is this because `course_subject_blah` may be populated with some garbage from apply before it is filled out? I don't think we need to do a join otherwise.